### ISSUE TITLE
ImagePredictor not splitting the dataset

### ISSUE THREAD

**Issue by zenetio** [awslabs/autogluon#1011]
I am running the example dog-breed-identification and when I run the statement below
`train_dataset, val_dataset, test_dataset = ImagePredictor.Dataset.from_folders(dataset_dir)`
For the returned value, I am getting all the images inside train_dataset variable and the other two variables, val and test dataset, are empty. I have the following structure
```data
    |-- breed-dog-identification
                   |-- images
                            |-- train
                                  |-- class1
                                  |-- class2
                                  |-- ...
```
Based on the documentation, I understand that the method should split the dataset into 3 parts.
Am I missing something here?
Tks.

**Comment by Innixma** [comment]
From what you have shown, your directory only includes train, so it would make sense to only have data in `train_dataset`. If the documentation says otherwise, can you please link to it?

**Comment by zenetio** [comment]
Hi @Innixma , thanks for your quick answer.
Based on the link below, I understand that autogluon should split the training dataset into, at least, two parts: train/val. But I the statement line I mentioned in my first post, returns just the train dataset managed in images and labels. No signal of val or test datasets.
https://auto.gluon.ai/dev/tutorials/image_prediction/kaggle.html

**Comment by Innixma** [comment]
@zhreshold could you take a look?

**Comment by zenetio** [comment]
Hi @Innixma, I just edited the source code and got the point. The function will receive the train directory and make the split process on the fly. The point is that the way how this process is commented in the documentation, lead us to think that the function will create train and val directories with splitted images.
I did spend many hours trying to understand and searching the internet looking for an answer why the code was not working.
So, I think that it should be good to update the docs explaining how or where the split process works.

**Comment by Innixma** [comment]
I see, thanks for the context. We will look into updating the docs to make it more clear.

**Comment by zhreshold** [comment]
- The default `from_folders` expects the train/val/test root folders to be pre-split into subdirectories similar to this example here: https://auto.gluon.ai/stable/tutorials/image_prediction/dataset.html#load-an-image-directory 
- If the structure of the folders are not designed to be the same way, there are still some convenient methods, e.g., use `from_folder` to load the whole folder regardless of train/val/test splits, and use pandas to split the `DataFrame` 

Yeah, we could improve the writings anyways.
